cmd/remove.go

```diff
@@ -72,7 +76,11 @@ func (o *RemoveOptions) Complete(f Factory, args []string) (err error) {
 		return err
 	}
 	if o.Refs, err = GetCurrentRefSelect(f, args, -1, nil); err != nil {
-		return err
+		// This error will be handled during validation
+		if err != repo.ErrEmptyRef {
+			return err
+		}
+		err = nil
```

Comment:
I initially removed the `Validate()` function and put its error here (it previously wasn’t getting called at all), but there is a test for it. I was trying not to make huge code changes here, so I put this together in a way that satisfies the existing tests. That said, I feel like keeping `Validate()` separate makes for more awkward code here: there’s a lot more checking in this spot, and if there is no repo ref (or it’s invalid), we will log that error *last* rather than first (because there are other validation-esque errors that come from `Complete()`).

Not sure whether the right solution is for other errors in `Complete()` to move to `Validate()`, to remove `Validate()` like I originally did, or something else.

Reply:
I like the answer that seems to be emerging from this PR: use cobra's built-ins more. 

This will be a bigger change, but I think the right answer is to call/adjust `Complete` methods into  [cobra pre-run hooks](https://github.com/spf13/cobra#prerun-and-postrun-hooks), and focus on defining all validation that can be done at the cmd level as `cobra.Command.Args` functions.